react/features/base/participants/components/Avatar.native.js

```diff
@@ -289,13 +279,36 @@ export default class Avatar extends Component<Props, State> {
      * @inheritdoc
      */
     render() {
-        const { source, useDefaultAvatar } = this.state;
+        const { source } = this.state;
 
         return (
             <Fragment>
                 { source && this._renderAvatar() }
-                { useDefaultAvatar && this._renderDefaultAvatar() }
+                { this.useDefaultAvatar && this._renderDefaultAvatar() }
             </Fragment>
         );
     }
 }
+
+/* eslint-disable react/no-multi-comp */
+
+/**
+ * Implements an avatar as a React Native/mobile {@link Component}.
+ *
+ * Note: we use `key` in order to trigger a new component creatin in case
+ * the URI changes.
+ */
+export default class Avatar extends PureComponent<Props> {
+    /**
+     * Implements React's {@link Component#render()}.
+     *
+     * @inheritdoc
+     */
+    render() {
+        return (
+            <AvatarContent
+                key = { this.props.uri }
```

Comment:
What is the benefit of using a key to force a new instance instead of working the component to update itself on uri change? Is it mostly for easier implementation?

Reply:
Yep. IMHO handling the props change is not worth it here, since the heavy-work is getting the image itself, so I went the easy route of re-creating the component.